deployer/src/persistence/mod.rs

```diff
@@ -71,19 +94,136 @@ impl Persistence {
         // longer present.
         let sqlite_options = SqliteConnectOptions::from_str(path)
             .unwrap()
-            .journal_mode(SqliteJournalMode::Wal);
+            .journal_mode(SqliteJournalMode::Wal)
+            // Set the ulid0 extension for converting UUIDs to ULID's in migrations.
+            // This uses the ulid0.so file in the crate root, with the
+            // LD_LIBRARY_PATH env set in build.rs.
+            .extension("ulid0");
 
         let pool = SqlitePool::connect_with(sqlite_options).await.unwrap();
 
-        Self::from_pool(pool).await
+        Self::configure(pool, resource_recorder_uri.to_string(), project_id).await
     }
 
-    #[allow(dead_code)]
+    #[cfg(test)]
     async fn new_in_memory() -> (Self, JoinHandle<()>) {
-        let pool = SqlitePool::connect("sqlite::memory:").await.unwrap();
+        let pool = SqlitePool::connect_with(
+            SqliteConnectOptions::from_str("sqlite::memory:")
+                .unwrap()
+                // Set the ulid0 extension for generating ULID's in migrations.
+                // This uses the ulid0.so file in the crate root, with the
+                // LD_LIBRARY_PATH env set in build.rs.
+                .extension("ulid0"),
+        )
+        .await
+        .unwrap();
         Self::from_pool(pool).await
     }
 
+    async fn configure(
```

Comment:
How is this method different from `from_pool`?

Reply:
Not much, it just accepts more arguments needed now by the `Persistence` struct. I deduped the code here... we could probably have a new struct created that wraps the `logs` senders too... but I don't find it extremely valuable atm. LMK what you think.